unity-ggjj/Assets/Scripts/TextDecoder/NarrativeScriptPlaylist.cs

```diff
@@ -0,0 +1,46 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class NarrativeScriptPlaylist : MonoBehaviour
+{
+    [field: Tooltip("List of narrative scripts to be played in order")]
+    [field: SerializeField] public List<NarrativeScript> NarrativeScripts { get; private set; }
+    
+    [field: Tooltip("List of narrative scripts to be used when taking penalties")]
+    [field: SerializeField] public List<NarrativeScript> FailureScripts { get; private set; }
+    
+    [field: Tooltip("A narrative script to be played on game over")]
+    [field: SerializeField] public NarrativeScript GameOverScript { get; private set; }
+    
+    private int _narrativeScriptIndex = -1;
+
+    /// <summary>
+    /// Initializes variables
+    /// </summary>
+    private void Awake()
+    {
+        NarrativeScripts.ForEach(script => script.Initialize());
+        FailureScripts.ForEach(script => script.Initialize());
+        GameOverScript.Initialize();
+    }
+
+    /// <summary>
+    /// Increments the narrative script index and returns the
+    /// corresponding script in the narrative scripts array
+    /// </summary>
+    /// <returns>The next narrative script in the array</returns>
+    public NarrativeScript GetNextNarrativeScript()
+    {
+        _narrativeScriptIndex++;
+        return _narrativeScriptIndex < NarrativeScripts.Count ? NarrativeScripts[_narrativeScriptIndex] : null;
```

Comment:
Nit: Haven't seen this on my first review pass unfortunately, but given that we're aiming for a chapter progression system (at least roughly) similar to what's outlined in https://github.com/Studio-Lovelies/GG-JointJustice-Unity/discussions/170, I think ending in a situation where the index exceeds the count of available scripts should probably throw an exception. It's an unexpected case we should never enter and -  if we do - have no way of recovering from. Does that work for you?
```suggestion
        if (_narrativeScriptIndex >= NarrativeScripts.Count)
        {
            throw new NotSupportedException("This playlist is on its last script already");
        }
        return NarrativeScripts[_narrativeScriptIndex];
```

Reply:
Corrected it's to its and committed suggestion

Reply:
Its/it's. It ALWAYS trips me up. Thanks for the fix. 👍